website/docs/generics.md

```diff
@@ -0,0 +1,1019 @@
+---
+id: generics
+title: Generic Methods and Classes
+sidebar_label: Generics
+---
+
+Sorbet has syntax for creating generic methods, classes, and interfaces.
+
+## Implementation status
+
+Generics are, by far, the most poorly supported feature within Sorbet. The
+implementation has many known, hard-to-fix bugs, which substantially limits
+their utility.
+
+It is hard to overstate how buggy Sorbet's support for generics is.
+
+We track known bugs in Sorbet's support for generics in the [Generics milestone]
+on Sorbet's issue tracker. Feel free to peruse the existing bugs, as well as
+report new bugs.
+
+Unfortunately, the abundance of bugs in Sorbet's generics implementation makes
+implementing sophisticated, generic abstractions in Sorbet exceedingly brittle.
+
+It is imperative to thoroughly test abstractions making use of Sorbet generics.
+
+The tests you'll need to write look substantially different from other Ruby
+tests you may be accustomed to writing, because these tests will have to live
+**at the type level** (statics), not at the expression level (runtime). These
+tests should account for two main failure modes:
```

Comment:
I feel like *should* is a bit overused here and the following sentences. Maybe you could it replace it with _need_ here?

Reply:
Hm, not sure. I think i'd prefer to leave it as is, but thanks for the suggestion.